x-pack/plugins/security_solution/public/common/hooks/use_app_toasts.ts

```diff
@@ -6,45 +6,79 @@
  */
 
 import { useCallback, useRef } from 'react';
+import { IEsError, isEsError } from '../../../../../../src/plugins/data/public';
 
 import { ErrorToastOptions, ToastsStart, Toast } from '../../../../../../src/core/public';
 import { useToasts } from '../lib/kibana';
-import { isAppError, AppError } from '../utils/api';
+import { isAppError } from '../utils/api';
 
 export type UseAppToasts = Pick<ToastsStart, 'addSuccess' | 'addWarning'> & {
   api: ToastsStart;
   addError: (error: unknown, options: ErrorToastOptions) => Toast;
 };
 
+/**
+ * This gives a better presentation of error data sent from the API (both general platform errors and app-specific errors).
+ * This uses platform's new Toasts service to prevent modal/toast z-index collision issues.
+ * This fixes some issues you can see with re-rendering since using a class such as notifications.toasts.
+ * This also has an adapter and transform for detecting if a bsearch's EsError is present and then adapts that to the
+ * Kibana error toaster model so that the network error message will be shown rather than a stack trace.
+ */
 export const useAppToasts = (): UseAppToasts => {
   const toasts = useToasts();
   const addError = useRef(toasts.addError.bind(toasts)).current;
   const addSuccess = useRef(toasts.addSuccess.bind(toasts)).current;
   const addWarning = useRef(toasts.addWarning.bind(toasts)).current;
 
-  const addAppError = useCallback(
-    (error: AppError, options: ErrorToastOptions) =>
-      addError(error, {
-        ...options,
-        toastMessage: error.body.message,
-      }),
-    [addError]
-  );
-
   const _addError = useCallback(
     (error: unknown, options: ErrorToastOptions) => {
-      if (isAppError(error)) {
-        return addAppError(error, options);
+      if (error != null && isEsError(error)) {
+        const err = esErrorToRequestError(error);
+        return addError(err, options);
+      } else if (isAppError(error)) {
+        return addError(error, options);
```

Comment:
One note @rylnd is that I chose to change this to use the default of `error.message` which is the toaster default to use if it is not given an explicit `toastMessage`.

I don't know if there was a reason to prefer using the `error.body.message` when it is an AppError for example that I'm not seeing?

Reply:
@FrankHassanabad the [old test](https://github.com/elastic/kibana/pull/97945/files/2b10a22f6f0265a5471353a2aab3b89a38843dab#diff-b6f76adce434d1d064bf67ae8c35a77c5f6a85b1ffcc8e3e743558710c02f818L43) seemed to indicate that `body.message` contained a more detailed message. Had you seen anything for/against that assertion?

Reply:
e.g. the `message` field was just a restatement of the status code, while `body.message` contained an actual reason.